packages/app/src/systems/Chains/eth/services/assetService.tsx

```diff
@@ -0,0 +1,53 @@
+import { isAddress } from 'viem';
+
+import type { BridgeAsset } from '~/systems/Bridge';
+import { db } from '~/systems/Core/utils/database';
+
+export type AssetServiceInputs = {
+  addAsset: { asset: BridgeAsset };
+  removeAsset: { address?: string };
+};
+
+export class AssetService {
+  static async upsertAsset(input: { data: BridgeAsset }) {
+    return db.transaction('rw!', db.assets, async () => {
+      const { address, ...updateData } = input.data;
+      const asset = await db.assets.get({ address });
+      if (asset) {
+        await db.assets.update(address || '', updateData);
+      } else {
+        await db.assets.add(input.data);
+      }
+
+      return db.assets.get({ address });
+    });
+  }
+
+  static async addAsset(input: AssetServiceInputs['addAsset']) {
+    return db.transaction('rw', db.assets, async () => {
+      await db.assets.add(input.asset);
+      return db.assets.get({ address: input.asset.address });
+    });
+  }
+
+  static async removeAsset({ address }: AssetServiceInputs['removeAsset']) {
+    try {
+      if (!isAddress(address || '')) {
+        throw new Error('Invlalid address');
+      }
+    } catch (error: unknown) {
+      return false;
+    }
```

Comment:
tis try is not needed, the objective is really throw the error instead of faking it with a `return false`